tests/phpunit/tests/test-canonical.php

```diff
@@ -232,6 +307,45 @@ public function test_category_from_plain_permalink() {
 		$this->assertCanonical( '?cat=' . self::$term_en, '/en/category/parent/' );
 	}
 
+	public function test_should_not_remove_query_string_parameter_from_category_rewritten_url() {
+		$this->assertCanonical( '/en/category/parent/?foo=bar', '/en/category/parent/?foo=bar' );
+	}
+
+	public function test_should_not_remove_query_string_parameter_from_tag_rewritten_url() {
+		$this->assertCanonical( '/en/tag/test-tag/?foo=bar', '/en/tag/test-tag/?foo=bar' );
+	}
+
+	public function test_custom_taxonomy_with_incorrect_language() {
+		$this->create_custom_term();
```

Comment:
This should not be useful to create the term in each test. But you need to register the taxonomy before creating the shared term.